Vivekananda_session3_4VP20CS062_Prajwal/Batman/web_scrapper.py

```diff
@@ -0,0 +1,103 @@
+import requests
+import re
+from bs4 import BeautifulSoup, element
+import psycopg2
+
+# For the credentials mentioned below, you may refer the docker-compose.yml present in myworld .
+db_name = 'member_db2'
+db_user = 'postgres'
+db_pass = '123456'
+db_host = 'psql-db2.0'
+db_port = '5432'
+        
+# This will create the connection the to postgres database.
+conn = psycopg2.connect(dbname=db_name, user=db_user, password=db_pass, host=db_host, port=db_port)
+        
+        
+def add_row_to_blog(title, date, time, author, content, recommended, html):
+    # This function will add the entry to database
+    sql = """INSERT INTO members_blog (title, release_date, blog_time, author,created_date,  content, recommended, html) VALUES (%s, %s::DATE,    		 %s::TIME, %s, NOW(), %s, %s, %s)"""
+            
+    with conn:
```

Comment:
@Prajwal7Amin why `with` is used?

Reply:
@Prajwal7Amin In your code i see that you've started transaction(insert/truncate) but I don't see a `commit` anywhere for it. 
Is the data actually stored in the database ? If yes, then how the data is stored?

Reply:
@anushds Yes, the data is stored in database.  Data is stored into 'members_blog' table and each row contains the data 
corresponding to the columns defined like title, date, author, content, blog time etc

Reply:
@Prajwal7Amin well you've answered my question partially. Could you talk about the `commit` that I've asked?

Reply:
@anushds Yes, I have understood it now. So, in my code I have not explicitly called 'commit' method on 'conn' object (conn.commit). Because ' with conn: ' block handles the transaction and automatically commits the changes when it exits. So ' with conn:' automatically commits the transaction when exiting the block.  This behavior is specified by the psycopg2 library.

Reply:
@Prajwal7Amin Good. The reason for this behaviour is because psycopg2 objects are context managers. When a error occurs while executing a query, the  transaction is automatically rolled back by the context manager. If you're wondering why rolling back a transaction is important then try to experiment with this. To do this, you've to enclose the `execute` line in a try-except block and ensure that the query breaks. Then continue to execute the next query. But make sure to print the exception otherwise you won't know what's going on.
Also I want you to look into converting the object of a class to context manager.